exercise/exercise_models.py

```diff
@@ -649,6 +653,14 @@ def _detect_submissions(self, profile, group):
             ))
         return False
 
+    def group_contains_staff(self, group, profile):
+        if group:
+            return any((
+                self.course_instance.is_course_staff(p.user)
```

Comment:
Does this allow the student to make new submissions with any group after the student has been added to a teacher's submission?

Reply:
No, the student is not able to make submissions after the teacher has added the student to the submission. The group_id is read from post data so I assume it is the group that made the submission which is currently in the validation process. But I have to admit that I don't know how the group made by teacher works, e.g. how is it saved in the database etc.

Reply:
I had a feeling that there is no group in the database since nobody has created the group when the teacher modifies the submitters afterwards in the "inspect submission" page.